crates/sui-graphql-client/src/lib.rs

```diff
@@ -145,6 +184,25 @@ impl Default for PaginationFilter<'_> {
     }
 }
 
+impl<T: Serialize> From<T> for Name {
+    fn from(value: T) -> Self {
+        Name(bcs::to_bytes(&value).unwrap())
+    }
+}
+
+impl From<BcsName> for Name {
+    fn from(value: BcsName) -> Self {
+        Name(value.0)
+    }
+}
+
+impl DynamicFieldOutput {
+    pub fn deserialize<T: DeserializeOwned>(&self) -> Result<T, anyhow::Error> {
+        let bcs = &self.name.bcs;
+        bcs::from_bytes::<T>(bcs).map_err(|_| anyhow!("Cannot decode BCS bytes"))
+    }
+}
```

Comment:
Do you want to have a way to check or validate the type before trying to deserialize? I think that would be the ideal, safest solution is to require someone to provide the type they're expecting and have the checked before trying to deserialize the value

Reply:
That sounds like a good extra check, fixed.